src/odemis/acq/test/orsay_milling_test.py

```diff
@@ -0,0 +1,282 @@
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import os
+import time
+import unittest
+from concurrent import futures
+
+from ConsoleClient.Communication.Connection import Connection
+
+import odemis
+from odemis.acq.orsay_milling import mill_rectangle
+from odemis.driver import orsay
+from odemis import model, util
+from odemis.util import testing
+from odemis.util import timeout
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_FIBVACUUM = {"name": "fib-vacuum", "role": "fib-vacuum"}
+CONFIG_FIBSOURCE = {"name": "fib-source", "role": "fib-source"}
+CONFIG_FIBBEAM = {"name": "fib-beam", "role": "fib-beam"}
+CONFIG_SCANNER = {"name": "scanner", "role": "scanner"}
+CONFIG_FOCUS = {"name": "focus", "role": "focus", "rng": (-2e-3, 2e-3)}
+CONFIG_DETECTOR = {"name": "detector", "role": "detector"}
+
+# Simulation:   192.168.56.101
+# Hardware:     192.168.30.101
+CONFIG_ORSAY = {"name": "Orsay", "role": "orsay", "host": "192.168.30.101",
+                "children": {"fib-vacuum": CONFIG_FIBVACUUM,
+                             "fib-source": CONFIG_FIBSOURCE,
+                             "fib-beam": CONFIG_FIBBEAM,
+                             "scanner": CONFIG_SCANNER,
+                             "focus": CONFIG_FOCUS,
+                             "detector": CONFIG_DETECTOR},
+                }
+
+NO_SERVER_MSG = "TEST_NOHW is set. No server to contact."
+CANT_FORCE_ACTUAL_MSG = "TEST_NOHW is not set to sim, cannot force data on Actual parameters of Orsay server outside " \
+                        "of simulation. "
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+MIMAS_CONFIG = CONFIG_PATH + "sim/mimas-sim.odm.yaml"
+
+# TEST_NOHW = os.environ.get("TEST_NOHW", "0")  # Default to Hw testing
+TEST_NOHW = "0"
+
+if TEST_NOHW == "0":
+    TEST_NOHW = False
+elif TEST_NOHW == "1":
+    CONFIG_ORSAY["host"] = "192.168.56.101"  # IP address of the simulated Orsay Physics Control Server
+    TEST_NOHW = True
+else:
+    raise ValueError("Unknown value of environment variable TEST_NOHW=%s" % TEST_NOHW)
+
+
+class TestMilling(unittest.TestCase):
+    """
+    Test milling functions
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW is True:
+            raise unittest.SkipTest("No hardware available.")
+
+        # start the backend
+        cls.scanner = model.getComponent(role="ion-beam")
+        cls.scanner.horizontalFoV.value = 100e-6  # m
+        cls.host = cls.scanner.parent.host
+
+    def _get_milling_state(self):
+        """
+        Gets current status of the milling process
+        :params miller: HybridPatternCreator in Orsay API
+        :return: current milling state
```

Comment:
As it's always "0" or "1", could you convert the return value to an int? It'll be nicer to read the callers.